spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mail/JndiMailAutoConfiguration.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2012-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.mail;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jndi.JndiLocatorDelegate;
+import org.springframework.mail.MailMessage;
+import org.springframework.util.StringUtils;
+
+import javax.mail.Session;
+import javax.naming.NamingException;
+
+/**
+ * @author Eddú Meléndez
+ */
+@Configuration
+@AutoConfigureBefore(MailSenderAutoConfiguration.class)
+@ConditionalOnClass(MailMessage.class)
+@ConditionalOnProperty(prefix = "spring.mail", name = "jndi-name")
+@EnableConfigurationProperties(MailProperties.class)
+public class JndiMailAutoConfiguration {
+
+	public static final String JNDI_NAME = "mail/Session";
+
+	@Autowired
+	private MailProperties properties;
+
+	@Bean
+	@ConditionalOnMissingBean
+	public Session session() throws NamingException {
+		if (StringUtils.hasLength(this.properties.getJndiName())) {
```

Comment:
I am confused. This auto-config only runs if `spring.mail.jndi-name` has a value and then you check if it has a value? The else part and default `JNDI_NAME` should go away.